### PR TITLE
Updated FreeBSD installation instructions with LLDB steps

### PR DESCRIPTION
**Pull request by ajensenwaud**
Included build instructions for FreeBSD on how to manually build and install LLDB on FreeBSD STABLE
Included FreeBSD LLDB install script

### CODE REVIEW

**Comment by josteink** [comment]
LGTM

**Comment by janhenke** [comment]
LGTM

File: src/pal/tools/freebsd-install-lldb.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+GIT=/usr/local/bin/git
+CMAKE=/usr/local/bin/cmake
+NINJA=/usr/local/bin/ninja
+PYTHON_PATH=/usr/local/include/python2.7
+
+$GIT clone http://llvm.org/git/llvm.git
+cd llvm/tools
+$GIT clone http://llvm.org/git/clang.git
+$GIT  clone http://llvm.org/git/lldb.git
```
Comment (jkotas) [inline src/pal/tools/freebsd-install-lldb.sh]: Cosmetic - extra space

File: Documentation/freebsd-instructions.md [hunk 2]
```diff
@@ -24,13 +24,39 @@ Install the following packages for the toolchain:
 - libunwind
 - gettext
 
-Note: LLDB is not yet available in the pkg repository on FreeBSD. The LLDB plugin is currently not being build on FreeBSD. Instructions for this will be added later.
-
 To install the packages you need:
 
-```janhenke@freebsd-frankfurt:~ % sudo pkg install bash cmake clang35 libunwind gettext```
+```
+janhenke@freebsd-frankfurt:~ sudo pkg install bash cmake clang35 libunwind gettext
```
Comment (jkotas) [inline Documentation/freebsd-instructions.md]: Undo this change to make the block consistent with the rest of the doc? (The single line and multiline versions look differently in the rendered view.)

File: Documentation/freebsd-instructions.md [hunk 3]
```diff
@@ -24,13 +24,39 @@ Install the following packages for the toolchain:
...
+
+Firstly, install the following packages: python ninja swig13 git (in addition to the packages above), i.e.
+
+```janhenke@freebsd-frankfurt:~ % sudo pkg install python ninja swig13 git```
+
+Then, run the install script in ~/coreclr/src/pal/tools: 
+
+```
+janhenke@freebsd-frankfurt:~ % ~/coreclr/src/pal/tools/freebsd-install-lldb.sh
+```
```
Comment (jkotas) [inline Documentation/freebsd-instructions.md]: Make it single line for consistency with the rest of the doc?

**Comment by jkotas** [comment]
LGTM modulo comments

**Comment by MattWhilden** [comment]
@ajensenwaud would you mind resolving the conflicts caused by the other cleanup PR that went in a few days ago? Looks good otherwise.

**Comment by jkotas** [comment]
LGTM. Could you please squash the commits into one?

File: Documentation/freebsd-instructions.md [hunk 4]
```diff
@@ -42,9 +64,7 @@ Build the Runtime
 
 To build the runtime on FreeBSD, run build.sh from the root of the coreclr repository:
 
-```
-janhenke@freebsd-frankfurt:~/git/coreclr % ./build.sh
-```
+```janhenke@freebsd-frankfurt:~/git/coreclr % ./build.sh```
```
Comment (tony) [inline Documentation/freebsd-instructions.md]: Use single ` for one-liners.

File: Documentation/freebsd-instructions.md [hunk 5]
```diff
@@ -24,13 +24,35 @@ Install the following packages for the toolchain:
 - libunwind
 - gettext
 
-Note: LLDB is not yet available in the pkg repository on FreeBSD. The LLDB plugin is currently not being build on FreeBSD. Instructions for this will be added later.
-
 To install the packages you need:
 
-```janhenke@freebsd-frankfurt:~ % sudo pkg install bash cmake clang35 libunwind gettext```
+``janhenke@freebsd-frankfurt:~ sudo pkg install bash cmake clang35 libunwind gettext```
```
Comment (tony) [inline Documentation/freebsd-instructions.md]: On a few projects I've worked on, using single tickets for inline/single lines helped avoid issues with some highlighters / md implementations.  https://github.com/dotnet/coreclr/pull/900

File: Documentation/freebsd-instructions.md [hunk 6]
```diff
@@ -24,13 +24,35 @@ Install the following packages for the toolchain:
...
+
+Note: This step is *optional* and is not required to build CoreCLR itself. If you intend on hacking or debugging the CoreCLR source code, you need to follow these steps. You must follow these steps *before* starting the build itself.
+
+In order to debug CoreCLR you will also need to install LLDB, the LLVM debugger. LLDB is still in the process of being ported to FreeBSD, so no official packages exist (see the FreeBSD LLDB Wiki page for more information on what has been ported so far: https://wiki.freebsd.org/lldb). However, it is possible to manually download and install LLDB from the LLVM source tree by following the instructions below: 
+
+Firstly, install the following packages: python ninja swig13 git (in addition to the packages above), i.e.
+
+```janhenke@freebsd-frankfurt:~ % sudo pkg install python ninja swig13 git```
+
+Then, run the install script in ~/coreclr/src/pal/tools:
```
Comment (tony) [inline Documentation/freebsd-instructions.md]: The instructions say assume `~/git/coreclr` and uses it in other places like `janhenke@freebsd-frankfurt:~/git/coreclr`. We should pick one or the other to stay consistent.

File: src/pal/tools/freebsd-install-lldb.sh [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
```
Comment (tony) [inline src/pal/tools/freebsd-install-lldb.sh]: Needs to be executable / ``chmod +x``

File: Documentation/freebsd-instructions.md [hunk 8]
```diff
@@ -24,13 +24,35 @@ Install the following packages for the toolchain:
...
+```janhenke@freebsd-frankfurt:~ % sudo pkg install python ninja swig13 git```
+
+Then, run the install script in ~/coreclr/src/pal/tools: 
 
-You now have all the required components.
+```janhenke@freebsd-frankfurt:~ % ~/coreclr/src/pal/tools/freebsd-install-lldb.sh```
+
+Note: LLDB will run su in order to install the LLDB build to /usr/local/include. 
+
+(Optional) If you wish to run sudo instead of su, you can change line 29-31 to:
```
Comment (tony) [inline Documentation/freebsd-instructions.md]: User has to be added to wheel group to run ``su``, ``pw groupmod wheel -m <user>``.

File: Documentation/freebsd-instructions.md (hunk 4 above)
Comment (josteink) [inline Documentation/freebsd-instructions.md]: This line *was* a one-liner but was nit-picked to use the same formatting as the rest of the document.

File: Documentation/freebsd-instructions.md (hunk 8 above)
Comment (josteink) [inline Documentation/freebsd-instructions.md]: I think we should pressume *some* platform-knowledge among the developers building and working with coreclr. I find it very unlikely that gaining root-access on your own local server is going to represent a problem for anyone who intends to work with this repo.

**Comment by josteink** [comment]
This PR is already almost one week old, with lots of nits and little *functional* changes.

I think at some point it is better to have *any* documentation or inconsistently formatted documentation over no documentation at all.

Sans the `chmod +x`issue, I vote for getting this commited, unless someone finds syntax or factual errors in the documentation.

Further improvements can always be delivered by new PRs.

**Comment by jkotas** [comment]
I will merge it in as soon as it is squashed into single commit.

**Comment by tony** [comment]
Still giving the script a QA on a new 10.1 install / checkout.   Awesome @ajensenwaud!  We should add this to FreeBSD wiki / make a handbook section / investigate making a port.

**Comment by tony** [comment]
Feel free to merge, by the way. I added those comments while I was QA'ing the build process. There are some more issues I'm working through.
